docs/developers/testing.md

```diff
@@ -0,0 +1,314 @@
+
+#### Running specific tests
+
+The `--tests` option allows the user to specify a space-delimited list of specific tests to run. Here is an example of how to use the option:
+```
+python run_core_tests.py \
+    --project-directory projects/OS-OS_VERSION/OS-OS_VERSION-DATABASE-DATABASE_VERSION \
+    --tests test_ils test_icp test_ihelp
+```
+This will deploy a single iRODS zone and execute the following tests in order on the catalog service provider: `test_ils`, `test_icp`, and `test_ihelp`. This can be used in conjunction with the `--concurrent-test-executors` option, but the tests will not be run in a deterministic order as the concurrent executors will take tests from the queue as they complete them.
+
+### How to run unit tests: `run_unit_tests.py`
+
+`run_unit_tests.py` is the main script for running the iRODS "unit" test suite (see [Unit test suite](#unit-test-suite-catch2)).
+
+Here is the minimal form for running the test suite for a particular `OS`, `OS_VERSION`, `DATABASE`, and `DATABASE_VERSION`:
+```
+python run_unit_tests.py --project-directory projects/OS-OS_VERSION/OS-OS_VERSION-DATABASE-DATABASE_VERSION
+```
+
+This will serially run the unit test suite using the latest release of the iRODS server. This will create a container for the iRODS server configured as the sole catalog service provider, a container for the database where the catalog is stored, and a network so that they can communicate as if it were a distributed environment. A running iRODS server is required because the Catch2 tests are not actually unit tests. Most of the same options available for `run_core_tests.py` are available for this script as well.
+
+The longest unit test takes about 2 minutes to complete.
+
+### How to stand up an iRODS zone: `stand_it_up.py`
+
+`stand_it_up.py` simply stands up an iRODS zone as specified. The minimal iRODS zone is a single catalog service provider and the database with which is communicates to maintain the catalog. This is the default behavior for this script.
+
+Here is the minimal form for creating a minimal iRODS zone using the latest release of the iRODS server for a particular `OS`, `OS_VERSION`, `DATABASE`, and `DATABASE_VERSION`:
+```
+python stand_it_up.py --project-directory projects/OS-OS_VERSION/OS-OS_VERSION-DATABASE-DATABASE_VERSION
+```
+
+#### Adding catalog service consumers
+
+The `--consumer-instance-count` option allows the user to specify how many catalog service consumers to deploy alongside the catalog service provider (of which there can only be one at this time). Here is an example of how to use the option:
+```
+python run_core_tests.py \
+    --project-directory projects/OS-OS_VERSION/OS-OS_VERSION-DATABASE-DATABASE_VERSION \
+    --consumer-instane-count 3
+```
+This will deploy a single iRODS zone with 1 catalog service provider, 1 database instance, and 3 catalog service consumers (5 containers).
```

Comment:
```suggestion
This will deploy a single iRODS zone with 1 catalog service provider, 1 database instance, and 3 catalog service consumers (a total of 5 containers).
```

Reply:
Done